tempesta_fw/http.c

```diff
@@ -4962,6 +4962,11 @@ tfw_http_get_ip_from_xff(TfwHttpReq *req)
 	 * IP address by the first value in X-Forwarded-For
 	 */
 	s_xff = req->h_tbl->tbl[TFW_HTTP_HDR_X_FORWARDED_FOR];
+	/*
+	 * Get only the first duplicate header
+	 */
+	if (TFW_STR_DUP(&s_xff))
+		s_xff = *__TFW_STR_CH(&s_xff, 0);
```

Comment:
`s_xff` should be a pointer to `TfwStr`, not `TfwStr`. The next function gets `const TfwStr *` as the argument and can't modify underlying data, thus there is no need to build a copy (twice) of XFF header in `s_xff` variable.